lib/pry/pry_instance.rb

```diff
@@ -575,27 +592,28 @@ def prompt_stack
 
   # Pushes the current prompt onto a stack that it can be restored from later.
   # Use this if you wish to temporarily change the prompt.
-  # @param [Array<Proc>] new_prompt
-  # @return [Array<Proc>] new_prompt
+  #
   # @example
-  #    new_prompt = [ proc { '>' }, proc { '>>' } ]
-  #    push_prompt(new_prompt) # => new_prompt
+  #   push_prompt(Pry::Prompt[:my_prompt])
+  #
+  # @param [Pry::Prompt] new_prompt
+  # @return [Pry::Prompt] new_prompt
   def push_prompt(new_prompt)
     prompt_stack.push new_prompt
   end
 
-  # Pops the current prompt off of the prompt stack.
-  # If the prompt you are popping is the last prompt, it will not be popped.
-  # Use this to restore the previous prompt.
-  # @return [Array<Proc>] Prompt being popped.
+  # Pops the current prompt off of the prompt stack. If the prompt you are
+  # popping is the last prompt, it will not be popped. Use this to restore the
+  # previous prompt.
+  #
   # @example
-  #    prompt1 = [ proc { '>' }, proc { '>>' } ]
-  #    prompt2 = [ proc { '$' }, proc { '>' } ]
-  #    pry = Pry.new :prompt => prompt1
-  #    pry.push_prompt(prompt2)
+  #    pry = Pry.new(prompt: Pry::Prompt[:my_prompt1])
+  #    pry.push_prompt(Pry::Prompt[:my_prompt2])
   #    pry.pop_prompt # => prompt2
   #    pry.pop_prompt # => prompt1
   #    pry.pop_prompt # => prompt1
+  #
+  # @return [Pry::Prompt] the prompt being popped
```

Comment:
Pry::Prompt is not returned, a Hash is, same for all other cases where Pry::Prompt is documented in this diff.
it really returns a Hash since that's still how we are choosing to represent a prompt.

Reply:
this is another reason to not use a Hash (for me). the docs would be better as a result too.